### PR TITLE
Production add & search & swagger_doc

### PR DESCRIPTION
**Pull request by William-Choe**
Complete two functions for production table: add and search.

1. Add: use the primary key of product table.
2. Search: use sequelize alias to achieve this function (modified in config/database.js), currently supports 11 operations.

### CODE REVIEW

**Comment by William-Choe** [comment]
New feature for the latest commit:
- Add relationship between draft project and production project
- Make sample unique with draft id and sample id
- Make search api available for latest timestamp and required timestamp

**Comment by William-Choe** [comment]
> Shall this be a integer type?

I will try it today

**Comment by tiandin** [comment]
> Shall this be a integer type?

File: amplify/backend/api/cmgdapi/src/models/ProjectProd.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      numberOfSamples: {
+        type: DataTypes.INTEGER,
+        allowNull: false,
+      },
+      assignee: {
+        type: DataTypes.STRING,
+        allowNull: false,
+      },
+      draftId: {
+        type: DataTypes.STRING,
```
Comment (tiandin) [inline amplify/backend/api/cmgdapi/src/models/ProjectProd.js:23]: I think we should directly comment on the code? @qwan820. What does the real data for draft Id look like?

File: amplify/backend/api/cmgdapi/src/controllers/production.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+          include: [
+            {
+              model: SampleProd,
+              as: "sampleProds"
+            }
+          ]
+      });
+      res.status(200, {'Content-Type': 'application/json'}).send();
+    } catch (err) {
+      console.log(err);
```
Comment (tiandin) [inline amplify/backend/api/cmgdapi/src/controllers/production.js:51]: Would be better don't leave the console.log in prod code. But I believe we do this many times before.

File: amplify/backend/api/cmgdapi/src/controllers/production.js [hunk 3]
```diff
@@ -48,7 +48,7 @@ module.exports = {
       });
       res.status(200, {'Content-Type': 'application/json'}).send();
     } catch (err) {
-      console.log(err);
+      // console.log(err);
```
Comment (tiandin) [inline amplify/backend/api/cmgdapi/src/controllers/production.js:51]: Why not just remove it?

File: amplify/backend/api/cmgdapi/src/controllers/production.js [hunk 4]
```diff
@@ -88,7 +88,7 @@ module.exports = {
 
       res.status(200, {'Content-Type': 'application/json'}).send(samples);
     } catch (err) {
-      console.log(err);
+      // console.log(err);
```
Comment (tiandin) [inline amplify/backend/api/cmgdapi/src/controllers/production.js:91]: Same here

File: amplify/backend/api/cmgdapi/src/controllers/production.js (hunk 3 above)
Comment (William-Choe) [reply amplify/backend/api/cmgdapi/src/controllers/production.js:51]: Maybe I will use it in future test, so should I remove them in the future commits?

Comment (tiandin) [reply amplify/backend/api/cmgdapi/src/controllers/production.js:51]: Yeah, you can add it whenever you want to do some further test. But we really shouldn't leave the unnecessary comment or console log in the prod code although we do this a lot in this project. So depends on you~

Comment (William-Choe) [reply amplify/backend/api/cmgdapi/src/controllers/production.js:51]: got it, I will leave them for the following commits.

**Review by tiandin — APPROVED** [review]
LGTM, approved

**Pull request merged**